### ISSUE TITLE
Uploading Set Information in the Sample tab

### ISSUE THREAD

**Issue by shekhjha** [ElucidataInc/ElMaven#228]
(1) Can we upload a standard CSV file in a predefined template that can update the "Set" column in the Sample tab?

(2) Can I sort by column "Set" and use the color scheme based on the "Set" values?

**Comment by sahil21** [comment]
@shekhjha 

`(1) Can we upload a standard CSV file in a predefined template that can update the "Set" column in the Sample tab?`

This feature has been implemented. The user can upload a CSV file containing set names for their samples which will update the “Set” column in the Samples widget. The CSV file should have two columns with the headers “Sample” and “Set” respectively (order of the columns does not matter).

![image](https://user-images.githubusercontent.com/21240516/27337865-5378cf34-55f1-11e7-90a9-a17578110730.png)

`(2) Can I sort by column "Set" and use the color scheme based on the "Set" values?`

All columns in the Sample window are sortable now. The user can now sort by “Set” and assign colors.

**Comment by chubukov** [comment]
@sahil21 we need a little more specification for both this and #296 

- what does the sample name need to look like? is extension (.mzxml) necessary, allowed, or neither?
- do you have to have a row for every sample?

I'm trying it out and getting the sort of unhelpful error message "trouble uploading set information"

**Comment by shubhra-agrawal** [comment]
@chubukov @shekhjha I'm attaching an example file [here](https://github.com/ElucidataInc/ElMaven/files/1213895/example_meta_file.csv.zip) which shows all allowed entries in a meta file.

- `what does the sample name need to look like? is extension (.mzxml) necessary, allowed, or neither?`

Sample name has to be the same as the sample file name. It is case-sensitive. Extensions are allowed but not necessary. 

- `do you have to have a row for every sample?`

No. You can have as many or as few rows in the meta file as you wish. 'Set', 'Scaling' or 'Injection Order' columns for a sample will remain empty in the project widget if the corresponding information is not provided in the meta file.

Additionally, the column headers are neither case nor order sensitive.

**Comment by chubukov** [comment]
@shubhra-agrawal Thanks -- not sure why my example wasn't working then,. I don't see the attachment you wanted to add -- can you re-post it?

**Comment by shubhra-agrawal** [comment]
@chubukov I've edited my comment to make the attachment more visible now. Clicking on "attached: example_file" will download the zipped file. Let me know if the template doesn't work for you. We'll work on more specific error messages for set information in a future sprint.

**Comment by chubukov** [comment]
@shubhra-agrawal Thanks -- I got it working, it turned out to be an excel issue.

**Comment by shubhra-agrawal** [comment]
@shekhjha 

> 2) Can I sort by column "Set" and use the color scheme based on the "Set" values?

I did not address the color scheme part before. We have decided to not implement this feature separately. What I would suggest is 
1. Sort the samples by "Set" column
2. Use Ctrl+Arrow keys to select all samples in a set
3. Use the "Change Sample Color" icon to assign a color to all samples in a cohort

I think all the points have been addressed so I am closing this issue.
